server/src/modules/clubs/mappers/club-map.ts

```diff
@@ -1,21 +1,78 @@
 import { UniqueEntityID } from '../../../shared/domain/unique-entity-id'
 import { ClubEntity } from '../../../shared/infra/db/entities/club.entity'
-import { Club } from '../domain/entities/club'
-import { ClubDTO } from './club-dto'
+import { EventEntity } from '../../../shared/infra/db/entities/event.entity'
+import { BasicEvent, Club } from '../domain/entities/club'
+
+interface BasicEventDTO {
+  name: string
+  startTime: string
+  endTime: string
+  backgroundImageURL?: string
+  tags: Array<string>
+}
+
+export interface ClubDTO {
+  name: string
+  description: string
+  backgroundImageURL?: string
+  iconURL: string
+  facebookLink?: string
+  twitterLink?: string
+  instagramLink?: string
+  websiteLink?: string
+  tags: Array<string>
+  events: Array<BasicEventDTO>
+}
+
+const basicEventMap = async (events: Array<EventEntity>): Promise<Array<BasicEvent>> => {
+  const eventPromises = events.map(async (event) => {
+    await event.tags.init()
+    return {
+      name: event.name,
+      startTime: event.startTime,
+      endTime: event.endTime,
+      backgroundImageURL: event.backgroundImageURL,
+      tags: event.tags.getItems().map((tag) => tag.name),
+    } as BasicEvent
+  })
+  return await Promise.all(eventPromises)
+}
 
 export class ClubMap {
   public static toDTO(club: Club): ClubDTO {
     return {
       name: club.name,
       description: club.description,
+      iconURL: club.iconURL,
+      backgroundImageURL: club.backgroundImageURL,
+      facebookLink: club.facebookLink,
+      twitterLink: club.twitterLink,
+      instagramLink: club.instagramLink,
+      websiteLink: club.websiteLink,
+      tags: club.tags,
+      events: club.events.map((event) => {
+        return {
+          ...event,
+          startTime: event.startTime.toJSON(),
+          endTime: event.endTime.toJSON(),
+        } as BasicEventDTO
+      }),
     }
   }
 
-  public static toDomain(clubEntity: ClubEntity): Club {
+  public static async toDomain(clubEntity: ClubEntity): Promise<Club> {
```

Comment:
random thought - every time we make a function async, we have to change a bunch of its usages across the codebase... I can't think of any way around this apart from making everything async by default

Reply:
yeah why are these methods `async`?

Reply:
in this case there's an async call in the function body to `basicEventMap`